doc/contribute/guidelines.rst

```diff
@@ -429,6 +431,8 @@ following exceptions:
   comment, ``//``, is not allowed.
 * Use ``/**  */`` for doxygen comments that need to appear in the documentation.
 
+When there are differences between the guidelines above and the rules enforced
+by the ``clang-format`` configuration, the guidelines above take precedence.
```

Comment:
Speaking for myself, I use clang-format not because I'm trying to avoid learning the coding style of the project, but I use clang-format so I can focus my development effort on material design decisions.  Using automated code formatting also allows reviewers to focus on the material design.

clang-format generally does a very good job handling the various line wrap conditions and is obviously much faster than manual edits.

Reply:
> clang-format generally does a very good job handling the various line wrap conditions and is obviously much faster than manual edits.

This is clearly not the case, in fact it is very bad on line wrapping and also fails badly at formatting multi-line macros. When reviewing the code, I will criticize most of the lines wrapped by clang-format. It is simply less readable, ugly, and more likely to be reformatted at every small change in the line.

Above, the mention of "rule enforcement" should be removed and rather replaced with a note that use of the clang format is entirely optional.

Reply:
@jfischer-no I would like this change to make it in before the RC3 hard freeze.  I would appreciate if you can revisit this.  Thanks.

Reply:
@jfischer-no  please resopond to the above or remove your change for request.